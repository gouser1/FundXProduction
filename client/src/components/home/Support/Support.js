import React from "react";
import {
  Grid,
  Box,
  Hidden,
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  Typography,
  Snackbar,
  Select,
  TextField,
} from "@material-ui/core";
import emailjs from "emailjs-com";
import MuiAlert from "@material-ui/lab/Alert";
import ContactImage from "../../../images/home/contact.svg";
import useStyles from "./SupportStyle";

function Alert(props) {
  return <MuiAlert elevation={6} variant="filled" {...props} />;
}

function Support(props) {
  const [open, setOpen] = React.useState(false);
  const [openSnackBar, setOpenSnackBar] = React.useState(false);

  const handleSnackBar = () => {
    setOpenSnackBar(true);
  };

  const handleSnackBarClose = (event, reason) => {
    if (reason === "clickaway") {
      return;
    }
    setOpenSnackBar(false);
  };

  const handleButtonClick = () => {
    setOpen(true);
  };

  const handleClose = (event, reason) => {
    if (reason === "clickaway") {
      return;
    }
    setOpen(false);
  };
  const classes = useStyles();

  const sendEmail = (e) => {
    e.preventDefualt();
    emailjs.sendForm(
      "service_97nbygu",
      "service_97nbygu",
      e.target,
      "user_8kEbzGq4FkNruvj1rGfFW"
    );
    e.target.reset();
  };

  return (
    <div className={classes.root}>
      <Grid
        container
        alignItems="center"
        justify="center"
        style={{ minHeight: "0vh" }}
      >
        <Grid item xs={12}>
          <Box className={classes.box} textAlign="center">
            <Typography className={classes.h1}>Ask us a Question</Typography>
          </Box>
        </Grid>
        <Grid item lg={4} md={12} sm={12} xs={12}>
          <Box className={classes.box}>
            <Typography className={classes.h2}>
              Need help with using FundX?
            </Typography>
          </Box>
          <Box>
            <Typography className={classes.p1}>
              Our Support team is always on call to help you with any questions
              or problems you may have. From technical support to bug issues, we
              are here. We also even offer a guided demo of the application if
              requested.
            </Typography>
          </Box>

          <Box pt={2} xs={12}>
            <Hidden only="xs">
              <Button
                onClick={() => handleButtonClick("/")}
                className={classes.button}
              >
                Contact us
              </Button>
            </Hidden>
            <Hidden smUp>
              <Box textAlign="center">
                <Button
                  variant="contained"
                  className={classes.button}
                  onClick={() => handleButtonClick("/")}
                >
                  Contact us
                </Button>
              </Box>
            </Hidden>

            <Dialog
              open={open}
              onClose={handleClose}
              aria-labelledby="form-dialog-title"
            >
              <form onSubmit={sendEmail}>
                <DialogTitle id="form-dialog-title">Contact Us</DialogTitle>
                <DialogTitle id="form-dialog-title">
                  What can we help you with?
                </DialogTitle>
                <DialogContent>
                  <Select
                    native
                    fullWidth
                    margin="dense"
                    id="category"
                    label="How can we help?"
                    autoFocus
                    name="helpOption"
                  >
                    <option aria-label="None" value="" />
                    <option value={10}>I need Technical Support</option>
                    <option value={10}>I need help with my Account</option>
                    <option value={20}>I'd like a Guided Demo</option>
                    <option value={30}>I'd like to Report a Bug</option>
                    <option value={30}>I'd like to Report a Legal Issue</option>
                    <option value={30}>I need help with Something Else</option>
                  </Select>
                  <TextField
                    margin="dense"
                    id="name"
                    label="Name"
                    type="name"
                    fullWidth
                    name="name"
                  />
                  <TextField
                    margin="dense"
                    id="email"
                    label="Email"
                    type="email"
                    fullWidth
                    name="email"
                  />
                  <TextField
                    id="message"
                    label="Message"
                    fullWidth
                    name="message"
                  />
                </DialogContent>
                <DialogActions>
                  <Button
                    type="submit"
                    value="Send Email"
                    onClick={handleClose}
                    color="#3bc693"
                  >
                    Cancel
                  </Button>

                  <Button
                    onClick={() => {
                      handleSnackBar();
                      handleClose();
                    }}
                    color="#3bc693"
                  >
                    Send
                  </Button>
                </DialogActions>
              </form>
            </Dialog>
          </Box>
        </Grid>
        <Grid item lg={4} md={12} sm={12} xs={12}>
          <Hidden mdDown>
            <Box pt={15} pl={10} justifyContent="center" display="flex">
              <img src={ContactImage} alt="" width="600px" height="300px" />
            </Box>
          </Hidden>
        </Grid>
      </Grid>

      <Snackbar
        open={openSnackBar}
        autoHideDuration={6000}
        onClose={handleSnackBarClose}
      >
        <Alert onClose={handleSnackBarClose} severity="success">
          Message Sent!
        </Alert>
      </Snackbar>
    </div>
  );
}

export default Support;
