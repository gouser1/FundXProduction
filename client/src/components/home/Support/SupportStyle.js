import { makeStyles } from "@material-ui/styles";

export default makeStyles(() => ({
  root: {
    flexGrow: 1,
    backgroundColor: "#3BC693",
    paddingLeft: "5%",
    paddingRight: "5%",
    paddingBottom: "10%",
  },
  typography: {
    fontFamily: "'Karla', sans-serif;",
    fontSize: "1.2em",
    color: "#46637f",
  },
  h1: {
    textAlign: "center",
    fontFamily: "'Karla', sans-serif;",
    fontSize: "1.5em",
    color: "white",
    fontWeight: "700",
    "@media (min-width:700px)": {
      fontSize: "3em",
    },
  },
  h2: {
    fontFamily: "'Karla', sans-serif;",
    fontSize: "1.6em",
    fontWeight: "700",
    color: "white",
    "@media (max-width:700px)": {
      fontSize: "1.4em",
      marginBottom: "5%",
      textAlign: "center",
    },
  },
  p1: {
    textAlign: "left",
    marginTop: "3%",
    fontFamily: "'Rubik', sans-serif;",
    fontSize: "1.2em",
    color: "white",
    letterSpacing: "0.02em",
    lineHeight: "1em",
    "@media (max-width:700px)": {
      fontSize: "1.3rem",
      textAlign: "center",
    },
    "@media (max-width:1400px)": {
      fontSize: "1.1rem",
    },
  },
  button: {
    fontFamily: "'Karla', sans-serif;",
    fontSize: "1.4em",
    color: "white",
    backgroundColor: "#0e5239",
    "&:hover": {
      backgroundColor: "#0e5239",
    },
    textTransform: "none",
  },

  box: {
    "@media (min-width:1500px)": {
      paddingTop: "4em",
    },
    "@media (max-width:1400px)": {
      paddingTop: "4em",
    },
    "@media (max-width:1000px)": {
      paddingTop: "5em",
    },
    "@media (max-width:800px)": {
      paddingTop: "4em",
    },
  },
}));
