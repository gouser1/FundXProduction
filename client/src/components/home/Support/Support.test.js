import React from "react";
import renderer from "react-test-renderer";

import Support from "./Support";

it("renders correctly", () => {
  const tree = renderer.create(<Support />).toJSON();
  expect(tree).toMatchSnapshot();
});
