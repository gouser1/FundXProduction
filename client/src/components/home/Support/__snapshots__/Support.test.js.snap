// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    style={
      Object {
        "minHeight": "0vh",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiBox-root MuiBox-root-8 makeStyles-box-7"
      >
        <p
          className="MuiTypography-root makeStyles-h1-3 MuiTypography-body1"
        >
          Ask us a Question
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-4"
    >
      <div
        className="MuiBox-root MuiBox-root-9 makeStyles-box-7"
      >
        <p
          className="MuiTypography-root makeStyles-h2-4 MuiTypography-body1"
        >
          Need help with using FundX?
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-10"
      >
        <p
          className="MuiTypography-root makeStyles-p1-5 MuiTypography-body1"
        >
          Our Support team is always on call to help you with any questions or problems you may have. From technical support to bug issues, we are here. We also even offer a guided demo of the application if requested.
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-11"
        xs={12}
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-4"
    />
  </div>
</div>
`;
